llvm/lib/DWARFLinker/Parallel/DWARFLinkerTypeUnit.cpp

```diff
@@ -287,12 +287,12 @@ uint32_t TypeUnit::addFileNameIntoLinetable(StringEntry *Dir,
   }
 
   uint32_t FileIdx = 0;
-  FilenamesMapTy::iterator FileEntry = FileNamesMap.find({FileName, DirIdx});
-  if (FileEntry == FileNamesMap.end()) {
+  auto [FileEntry, Inserted] = FileNamesMap.try_emplace({FileName, DirIdx});
+  if (Inserted) {
     // We currently do not support more than UINT32_MAX files.
     assert(LineTable.Prologue.FileNames.size() < UINT32_MAX);
     FileIdx = LineTable.Prologue.FileNames.size();
-    FileNamesMap.insert({{FileName, DirIdx}, FileIdx});
+    FileEntry->second = FileIdx;
```

Comment:
Maybe it would make sense to pass the new index directly to try_emplace?
```
auto [FileEntry, Inserted] = FileNamesMap.try_emplace({FileName, DirIdx}, LineTable.Prologue.FileNames.size());
uint32_t FileIdx = FileEntry->second;
```
No strong opinion though.

Reply:
I just posted https://github.com/llvm/llvm-project/pull/109670 as a follow-up.